sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests/FunctionalTests/AzureMonitorTestBase.cs

```diff
@@ -0,0 +1,167 @@
+        /// <code>var creds = new ApiKeyClientCredentials("00000000-0000-0000-0000-000000000000");</code>
+        /// </remarks>
+        protected async Task<ApplicationInsightsDataClient> GetApplicationInsightsDataClientAsync()
+        {
+            var clientId = TestEnvironment.ClientId;
+            var clientSecret = TestEnvironment.ClientSecret;
+            var domain = TestEnvironment.TenantId;
+            var authEndpoint = "https://login.microsoftonline.com";
+            var tokenAudience = "https://api.applicationinsights.io/";
+            var adSettings = new ActiveDirectoryServiceSettings
+            {
+                AuthenticationEndpoint = new Uri(authEndpoint),
+                TokenAudience = new Uri(tokenAudience),
+                ValidateAuthority = true
+            };
+
+            var creds = await ApplicationTokenProvider.LoginSilentAsync(domain, clientId, clientSecret, adSettings);
+
+            var handler = new HttpPipelineMessageHandler(new HttpPipeline(Recording.CreateTransport(new HttpClientTransport())));
+            var httpClient = new HttpClient(handler);
+
+            var client = new ApplicationInsightsDataClient(credentials: creds, httpClient: httpClient, disposeHttpClient: true);
+            return client;
+        }
+
+        /// <summary>
+        /// Application Insights Ingestion is not immediate.
+        /// On a good day this can still take a few minutes.
+        /// - IN LIVE OR RECORD TEST MODE
+        /// Tests must wait before querying for telemetry.
+        /// This test will wait for a short period, up to a timeout duration.
+        /// - IN PLAYBACK TEST MODE
+        /// Test must repeat the same behavior of the recording.
+        /// In this scenario there is a negligible delay.
+        /// </summary>
+        /// <remarks>
+        /// TODO: WHEN WE INTRODUCE MORE TESTS, WILL NEED TO ADD PARAMETERS HERE TO CUSTOMIZE THE QUERY.
+        /// </remarks>
+        protected async Task<IList<EventsTraceResult>> FetchTelemetryAsync()
+        {
+            var timeoutDuration = TimeSpan.FromMinutes(5); // timeout after 5 minutes.
+            var period = GetWaitPeriod(); // query once every 30 seconds.
+
+            var client = await this.GetApplicationInsightsDataClientAsync();
+            IList<EventsTraceResult> telemetry = null;
+
+            for (double actualDuration = 0; actualDuration <= timeoutDuration.TotalMilliseconds; actualDuration += period.TotalMilliseconds)
+            {
+                await Task.Delay(period);
+
+                var queryResult = await client.Events.GetTraceEventsAsync(appId: TestEnvironment.ApplicationId, timespan: QueryDuration.TenMinutes);
+
+                if (queryResult.Value.Any())
+                {
+                    telemetry = queryResult.Value;
+                    break;
+                }
+            }
+
+            if (telemetry == null)
```

Comment:
Hm, how do we know the library keeps working if the test never fails?

Reply:
Sorry, i don't understand your comment.

At this point in the test, the Exporter has finished its job. 
Here we're dependent on the Ingestion service and the ApplicationInsighst.Query library. 
If the Query returns 0 telemetry items, I would expect a delay in the Ingestion service.

I'm calling `Assert.Inconclusive` which I would expect to fail the build. But i can change this to `Assert.IsNotNull` instead if that's more appropriate.

Reply:
>I'm calling Assert.Inconclusive which I would expect to fail the build

Nope, it doesn't. That's why I was wondering. Inconclusive would just log a warning so we would never catch when the test starts failing.